From d33cff2c731416bdc625d9188f9f94c9d09171c0 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:24 -0400
Subject: [tools] perf: Adapt the TASK_STATE_TO_CHAR_STR to new value in kernel space

Message-id: <1410181752-25631-200-git-send-email-jolsa@redhat.com>
Patchwork-id: 91816
O-Subject: [PATCH RHEL7.1 BZ1134356 199/307] perf tools: Adapt the TASK_STATE_TO_CHAR_STR to new value in kernel space.
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit e936e8e459e14af7432a775f8139a79b71e41afc
Author: Dongsheng <yangds.fnst@cn.fujitsu.com>
Date: Mon May 5 16:05:54 2014 +0900

description
===========
Currently, TASK_STATE_TO_CHAR_STR in kernel space is already expanded to RSDTtZXxKWP,
but it is still RSDTtZX in perf sched tool.

This patch update TASK_STATE_TO_CHAR_STR to the new value in kernel space.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-sched.c b/tools/perf/builtin-sched.c
index 7eae501..4f0dd21 100644
--- a/tools/perf/builtin-sched.c
+++ b/tools/perf/builtin-sched.c
@@ -66,7 +66,7 @@ struct sched_atom {
  struct task_desc *wakee;
 };
 
-#define TASK_STATE_TO_CHAR_STR "RSDTtZX"
+#define TASK_STATE_TO_CHAR_STR "RSDTtZXxKWP"
 
 enum thread_state {
  THREAD_SLEEPING = 0,
-- 
1.7.1